packages/extractor-vue/src/vue-extractor.ts

```diff
@@ -0,0 +1,46 @@
+import * as vueCompiler from "@vue/compiler-sfc"
+// @ts-expect-error issue in typings, will be fixed shortly
+import babel from "@lingui/cli/api/extractors/babel"
+import { ExtractorCtx, ExtractorType } from "@lingui/conf"
+import { DEFAULT_EXTENSIONS } from "@babel/core"
+
+const babelRe = new RegExp(
+  "\\.(" +
+    [...DEFAULT_EXTENSIONS, ".ts", ".mts", ".cts", ".tsx", ".vue"]
+      .map((ext) => ext.slice(1))
+      .join("|") +
+    ")$",
+  "i"
+)
+
+export const vueExtractor: ExtractorType = {
+  match(filename: string) {
+    return babelRe.test(filename)
+  },
```

Comment:
I think you can implement this extractor strictly for vue files and left rest for the original extractor: 

```ts
match(filename: string) {
  return filename.endsWith(".vue")
},
```

```ts
import { vueExtractor } from "@lingui/extractor-vue"
import babel from "@lingui/cli/api/extractors/babel"

const linguiConfig = {
  locales: ["en", "nb"],
  sourceLocale: "en",
  catalogs: [
    {
      path: "<rootDir>/src/{locale}",
      include: ["<rootDir>/src"],
    },
  ],
  extractors: [babel, vueExtractor],
}
export default linguiConfig
```